### DISCUSSION TITLE
Simple Navigation system doesn't seem to work on android & Wasm?

### DISCUSSION THREAD

**Discussion by Mrcubix** [AvaloniaUI/Avalonia#8993] [Q&A]
minimal demo repos: 
- Android: https://github.com/Mrcubix/Avalonia-Router-Test/tree/android
- Web: https://github.com/Mrcubix/Avalonia-Router-Test/tree/Web

I'm trying to get a navigation system working on every platform right now and noticed that for some reason it doesn't seem to work on android and wasm

Android :

First of all, whenever i start the app, the app freeze until i tap on certain parts of the background?
Second, when i tap on the button, nothing happen, except this get printed in logcat `fb_post fps = 0.199704`
Finally, the first issue happen again if the app isn't in focus anymore (in hibernation, in another app, ...)

Looking into things a bit more, i noticed this getting spammed:

```log
09-19 20:22:37.190   718   777 E BufferQueueProducer: [SurfaceView - com.CompanyName.router_test/crc646a1044cb1c7f6fdb.MainActivity] connect: already connected (cur=1 req=1)
09-19 20:22:37.191 12566 12584 E libEGL  : eglCreateWindowSurface: native_window_api_connect (win=0x94851008) failed (0xffffffea) (already connected to another API?)
09-19 20:22:37.191 12566 12584 E libEGL  : eglCreateWindowSurface:482 error 3003 (EGL_BAD_ALLOC)
```

[log.txt](https://github.com/AvaloniaUI/Avalonia/files/9601627/log.txt) (also available on the repo)

this file should contain most if not all of the important logs for the app, kind of complicated to get something filtered.

Wasm:

The app start fine, it doesn't hang, the button in the middle that supposed to lead me to the list page is just not working
No errors in the F12 console, only thse logs:

```
starting html canvas setup
blazor.webassembly.js:1 Interop created
blazor.webassembly.js:1 jsglinfo created - init gl
blazor.webassembly.js:1 create glcontext
blazor.webassembly.js:1 glcontext created and resource limit set
```

The navigation system works on windows.
Tested on Android 7.1.2 (API 21) & Android 8.0.0 (API 26) (Android), Windows for Wasm

## Update

It seems as you can't debug an app in blazor if your project path contains a # character, that has been taken care of using an hard link, nw i'm able to debug it properly.

First thing i noticed, the function responsible for changing the current view, work as intended, it change the `CurrentViewModel` variable to whichever one i specify.

Now, i have to figure out why blazor won't change the current view.

**Comment by maxkatz6** [comment]
There should be zero differences how binding and in-app navigation works in Android/WASM and desktop.
But you need to remember that mobile and web are **single-view** platforms without windows support. Which means, your MainWindow is never used there.

In your code there is a OnFrameworkInitializationCompleted, where you create MainView as a root for the mobile and web.
And I don't see if MainView has a navigation inside. Just a simple button to trigger navigation.

> It seems as you can't debug an app in blazor if your project path contains a # character, that has been taken care of using an hard link, nw i'm able to debug it properly.

Please open an issue in dotnet/runtime or aspnetcore repository (not sure which one is responsible for wasm debugging).

**Reply by Mrcubix** [reply]
Surprised to see that only now since i spent about 2 days looking into it, i would had expected to read this on one of the 50+ different articles and documentation pages i visited.

But i guess i wasn't asking the right question.

Guess i'll have to read more about it to see how they make it work now that i know that

>>It seems as you can't debug an app in blazor if your project path contains a # character, that has been taken care of using an hard link, >>nw i'm able to debug it properly.
>
>Please open an issue in dotnet/runtime or aspnetcore repository (not sure which one is responsible for wasm debugging).

It seem more likely to be a browser issue more than anything, the path should had been url encoded internally but apparently, it wasn't done for local files
